### PR TITLE
fix for 'show vlan id {vlan_id}' parsing.

### PR DESCRIPTION
**Pull request by viktklass**
## Description
fixed regex for 'show vlan id {vlan_id}' parsing to be more inclusive.

## Motivation and Context
[related issue](https://github.com/CiscoTestAutomation/genieparser/issues/584)

Regex for show vlan id couldn't read the symbol '-', '_' and ' '. 
Changed regex to match any vlan name
Regex for vlan status is set for the 4 possible statuses.
regex for ports allows more possible combination according to the available ones.

also  added 2 new test for vlan-name containing blankspace and dash-symbol.

## Screenshots:
```
python folder_parsing_job.py -o iosxe -c ShowVlanId 
2021-12-07T13:45:34: %AETEST-INFO: +------------------------------------------------------------------------------+
2021-12-07T13:45:34: %AETEST-INFO: |                   Starting testcase SuperFileBasedTesting                    |
2021-12-07T13:45:34: %AETEST-INFO: +------------------------------------------------------------------------------+
2021-12-07T13:45:34: %AETEST-INFO: +------------------------------------------------------------------------------+
2021-12-07T13:45:34: %AETEST-INFO: |                            Starting section setup                            |
2021-12-07T13:45:34: %AETEST-INFO: +------------------------------------------------------------------------------+
2021-12-07T13:45:34: %AETEST-INFO: The result of section setup is => PASSED
2021-12-07T13:45:34: %AETEST-INFO: The result of testcase SuperFileBasedTesting is => PASSED
2021-12-07T13:45:34: %AETEST-INFO: +------------------------------------------------------------------------------+
2021-12-07T13:45:34: %AETEST-INFO: |                           Starting testcase iosxe                            |
2021-12-07T13:45:34: %AETEST-INFO: +------------------------------------------------------------------------------+
2021-12-07T13:45:34: %AETEST-INFO: +------------------------------------------------------------------------------+
2021-12-07T13:45:34: %AETEST-INFO: |                            Starting section setup                            |
2021-12-07T13:45:34: %AETEST-INFO: +------------------------------------------------------------------------------+
2021-12-07T13:45:34: %AETEST-INFO: The result of section setup is => PASSED
2021-12-07T13:45:34: %AETEST-INFO: +------------------------------------------------------------------------------+
2021-12-07T13:45:34: %AETEST-INFO: |                         Starting section ShowVlanId                          |
2021-12-07T13:45:34: %AETEST-INFO: +------------------------------------------------------------------------------+
2021-12-07T13:45:34: %AETEST-INFO: +..............................................................................+
2021-12-07T13:45:34: %AETEST-INFO: :                     Starting STEP 1: iosxe -> ShowVlanId                     :
2021-12-07T13:45:34: %AETEST-INFO: +..............................................................................+
2021-12-07T13:45:34: %AETEST-INFO: +..............................................................................+
2021-12-07T13:45:34: %AETEST-INFO: :            Starting STEP 1.1: Test Golden -> iosxe -> ShowVlanId             :
2021-12-07T13:45:34: %AETEST-INFO: +..............................................................................+
2021-12-07T13:45:34: %AETEST-INFO: +..............................................................................+
2021-12-07T13:45:34: %AETEST-INFO: :     Starting STEP 1.1.1: Gold -> iosxe -> ShowVlanId -> golden_output_1      :
2021-12-07T13:45:34: %AETEST-INFO: +..............................................................................+
2021-12-07T13:45:34: %AETEST-INFO: The result of STEP 1.1.1: Gold -> iosxe -> ShowVlanId -> golden_output_1 is => PASSED
2021-12-07T13:45:34: %AETEST-INFO: +..............................................................................+
2021-12-07T13:45:34: %AETEST-INFO: :     Starting STEP 1.1.2: Gold -> iosxe -> ShowVlanId -> golden_output_2      :
2021-12-07T13:45:34: %AETEST-INFO: +..............................................................................+
2021-12-07T13:45:34: %AETEST-INFO: The result of STEP 1.1.2: Gold -> iosxe -> ShowVlanId -> golden_output_2 is => PASSED
2021-12-07T13:45:34: %AETEST-INFO: +..............................................................................+
2021-12-07T13:45:34: %AETEST-INFO: :     Starting STEP 1.1.3: Gold -> iosxe -> ShowVlanId -> golden_output_3      :
2021-12-07T13:45:34: %AETEST-INFO: +..............................................................................+
2021-12-07T13:45:34: %AETEST-INFO: The result of STEP 1.1.3: Gold -> iosxe -> ShowVlanId -> golden_output_3 is => PASSED
2021-12-07T13:45:34: %AETEST-INFO: The result of STEP 1.1: Test Golden -> iosxe -> ShowVlanId is => PASSED
2021-12-07T13:45:34: %AETEST-INFO: +..............................................................................+
2021-12-07T13:45:34: %AETEST-INFO: :             Starting STEP 1.2: Test Empty -> iosxe -> ShowVlanId             :
2021-12-07T13:45:34: %AETEST-INFO: +..............................................................................+
2021-12-07T13:45:34: %AETEST-INFO: +..............................................................................+
2021-12-07T13:45:34: %AETEST-INFO: :      Starting STEP 1.2.1: Empty -> iosxe -> ShowVlanId -> empty_output       :
2021-12-07T13:45:34: %AETEST-INFO: +..............................................................................+
2021-12-07T13:45:34: %AETEST-INFO: The result of STEP 1.2.1: Empty -> iosxe -> ShowVlanId -> empty_output is => PASSED
2021-12-07T13:45:34: %AETEST-INFO: The result of STEP 1.2: Test Empty -> iosxe -> ShowVlanId is => PASSED
2021-12-07T13:45:34: %AETEST-INFO: The result of STEP 1: iosxe -> ShowVlanId is => PASSED
2021-12-07T13:45:34: %AETEST-INFO: +..........................................................+
2021-12-07T13:45:34: %AETEST-INFO: :                       STEPS Report                       :
2021-12-07T13:45:34: %AETEST-INFO: +..........................................................+
2021-12-07T13:45:34: %AETEST-INFO: STEP 1 - iosxe -> ShowVlanId                      Passed    
2021-12-07T13:45:34: %AETEST-INFO: STEP 1.1 - Test Golden -> iosxe -> ShowVlanId     Passed    
2021-12-07T13:45:34: %AETEST-INFO: STEP 1.1.1 - Gold -> iosxe -> ShowVlanId -> golden_output_1Passed    
2021-12-07T13:45:34: %AETEST-INFO: STEP 1.1.2 - Gold -> iosxe -> ShowVlanId -> golden_output_2Passed    
2021-12-07T13:45:34: %AETEST-INFO: STEP 1.1.3 - Gold -> iosxe -> ShowVlanId -> golden_output_3Passed    
2021-12-07T13:45:34: %AETEST-INFO: STEP 1.2 - Test Empty -> iosxe -> ShowVlanId      Passed    
2021-12-07T13:45:34: %AETEST-INFO: STEP 1.2.1 - Empty -> iosxe -> ShowVlanId -> empty_outputPassed    
2021-12-07T13:45:34: %AETEST-INFO: ............................................................
2021-12-07T13:45:34: %AETEST-INFO: The result of section ShowVlanId is => PASSED
2021-12-07T13:45:34: %AETEST-INFO: +------------------------------------------------------------------------------+
2021-12-07T13:45:34: %AETEST-INFO: |                           Starting section cleanup                           |
2021-12-07T13:45:34: %AETEST-INFO: +------------------------------------------------------------------------------+
2021-12-07T13:45:34: %AETEST-INFO: The result of section cleanup is => PASSED
2021-12-07T13:45:34: %AETEST-INFO: The result of testcase iosxe is => PASSED
2021-12-07T13:45:34: %GENIE-INFO: +------------------------------------------------------------------------------+
2021-12-07T13:45:34: %GENIE-INFO: |                               Unittest results                               |
2021-12-07T13:45:34: %GENIE-INFO: +------------------------------------------------------------------------------+
2021-12-07T13:45:34: %GENIE-INFO:  SECTIONS/TESTCASES                                                      RESULT   
2021-12-07T13:45:34: %GENIE-INFO: --------------------------------------------------------------------------------
2021-12-07T13:45:34: %GENIE-INFO: .
2021-12-07T13:45:34: %GENIE-INFO: |-- SuperFileBasedTesting                                                 PASSED
2021-12-07T13:45:34: %GENIE-INFO: |   `-- setup                                                             PASSED
2021-12-07T13:45:34: %GENIE-INFO: `-- iosxe                                                                 PASSED
2021-12-07T13:45:34: %GENIE-INFO:     |-- setup                                                             PASSED
2021-12-07T13:45:34: %GENIE-INFO:     |-- ShowVlanId                                                        PASSED
2021-12-07T13:45:34: %GENIE-INFO:     |   |-- Step 1: iosxe -> ShowVlanId                                   PASSED
2021-12-07T13:45:34: %GENIE-INFO:     |   |-- Step 1.1: Test Golden -> iosxe -> ShowVlanId                  PASSED
2021-12-07T13:45:34: %GENIE-INFO:     |   |-- Step 1.1.1: Gold -> iosxe -> ShowVlanId -> golden_output_1    PASSED
2021-12-07T13:45:34: %GENIE-INFO:     |   |-- Step 1.1.2: Gold -> iosxe -> ShowVlanId -> golden_output_2    PASSED
2021-12-07T13:45:34: %GENIE-INFO:     |   |-- Step 1.1.3: Gold -> iosxe -> ShowVlanId -> golden_output_3    PASSED
2021-12-07T13:45:34: %GENIE-INFO:     |   |-- Step 1.2: Test Empty -> iosxe -> ShowVlanId                   PASSED
2021-12-07T13:45:34: %GENIE-INFO:     |   `-- Step 1.2.1: Empty -> iosxe -> ShowVlanId -> empty_output      PASSED
2021-12-07T13:45:34: %GENIE-INFO:     `-- cleanup                                                           PASSED
2021-12-07T13:45:34: %GENIE-INFO: +------------------------------------------------------------------------------+
2021-12-07T13:45:34: %GENIE-INFO: |                                   Summary                                    |
2021-12-07T13:45:34: %GENIE-INFO: +------------------------------------------------------------------------------+
2021-12-07T13:45:34: %GENIE-INFO:  Number of ABORTED                                                            0 
2021-12-07T13:45:34: %GENIE-INFO:  Number of BLOCKED                                                            0 
2021-12-07T13:45:34: %GENIE-INFO:  Number of ERRORED                                                            0 
2021-12-07T13:45:34: %GENIE-INFO:  Number of FAILED                                                             0 
2021-12-07T13:45:34: %GENIE-INFO:  Number of PASSED                                                             2 
2021-12-07T13:45:34: %GENIE-INFO:  Number of PASSX                                                              0 
2021-12-07T13:45:34: %GENIE-INFO:  Number of SKIPPED                                                            0 
2021-12-07T13:45:34: %GENIE-INFO:  Total Number                                                                 2 
2021-12-07T13:45:34: %GENIE-INFO:  Success Rate                                                            100.0% 
2021-12-07T13:45:34: %GENIE-INFO: --------------------------------------------------------------------------------
2021-12-07T13:45:34: %GENIE-INFO:  Total Passing Unittests                                                      4 
2021-12-07T13:45:34: %GENIE-INFO:  Total Failed Unittests                                                       0 
2021-12-07T13:45:34: %GENIE-INFO:  Total Errored Unittests                                                      0 
2021-12-07T13:45:34: %GENIE-INFO:  Total Unittests                                                              4 
2021-12-07T13:45:34: %GENIE-INFO: --------------------------------------------------------------------------------
```

## Checklist:

- [x] I have updated the changelog.
- [not applicable] I have updated the documentation (If applicable).
- [x] I have added tests to cover my changes (If applicable).
- [x] All new and existing tests passed.
- [x] All new code passed compilation.

### CODE REVIEW

File: changelog/undistributed/changelog_update_iosxe_show_vlan_id_20211020.rst [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+--------------------------------------------------------------------------------
+* IOSXE
+    * Modified ShowVlanId
+        * Added support vlan-name to be more diverse, including " ", "-", "_"
+
+--------------------------------------------------------------------------------
+                            New
+--------------------------------------------------------------------------------
+
+
```
Comment (lsheikal) [inline changelog/undistributed/changelog_update_iosxe_show_vlan_id_20211020.rst:12]: Kindly remove these lines. Changes should be only added under New block when there is a new feature.

**Review by lsheikal — APPROVED** [review]
Approved with comments.

**Comment by viktklass** [comment]
Hi!
Is there something waiting on me now in this case?

//Github-n00b

**Review by ThomasJRyan — APPROVED** [review]

**Comment by ThomasJRyan** [comment]
My apologies for taking so long @viktklass . It looks good. Thank you for the contribution

**Pull request merged**